src/remoteConnector.ts

```diff
@@ -622,6 +619,14 @@ export default class RemoteConnector extends Disposable {
 		throw new Error('SSH password modal dialog, Canceled');
 	}
 
+	private getGitpodSession() {
```

Comment:
Should not it accept gitpodHost and always do a check 
```
if (new URL(params.gitpodHost).host !== new URL(gitpodHost).host) {
    ...
```
to be sure that we don't use session for wrong host